test/functional/wallet_v3_txs.py

```diff
@@ -0,0 +1,560 @@
+        outputs = {self.alice.getnewaddress() : alice_unspent['amount'] - Decimal(0.00000120)} # two outputs
+        alice_tx = self.alice.createrawtransaction(inputs=inputs, outputs=outputs) # don't set the version here
+
+        assert_raises_rpc_error(
+            -4,
+            "Can't spend unconfirmed version 3 pre-selected input with a version 2 tx",
+            self.alice.fundrawtransaction,
+            alice_tx
+        )
+
+    @cleanup
+    def v3_tx_evicted_from_mempool_by_sibling(self):
+        # unconfirmed v3 tx to alice & bob
+        self.log.info("Test v3 transaction with conflicting sibling")
+        self.generate(self.nodes[2], 1)
+
+        inputs=[]
+        outputs = {self.alice.getnewaddress() : 2.0, self.bob.getnewaddress() : 2.0}
+        parent_txid = self.send_tx(self.charlie, inputs, outputs, 3)
+
+        # alice spends her output with a v3 transaction
+        alice_unspent = self.alice.listunspent(minconf=0)[0]
+        inputs=[{'txid' : parent_txid, 'vout' : alice_unspent['vout']},]
+        outputs = {self.alice.getnewaddress() : alice_unspent['amount'] - Decimal(0.00000120)} # two outputs
+        alice_txid = self.send_tx(self.alice, inputs, outputs, 3)
+
+        # bob tries to spend money
+        bob_unspent = self.bob.listunspent(minconf=0)[0]
+        inputs=[{'txid' : parent_txid, 'vout' : bob_unspent['vout']},]
+        outputs = {self.bob.getnewaddress() : bob_unspent['amount'] - Decimal(0.00010120)} # two outputs
+        bob_txid = self.send_tx(self.bob, inputs, outputs, 3)
+
+        assert_equal(self.alice.gettransaction(alice_txid)['mempoolconflicts'], [bob_txid])
+
+        self.log.info("Test that re-submitting Alice's transaction with a higher fee removes bob's tx as a mempool conflict")
+        inputs=[{'txid' : parent_txid, 'vout' : alice_unspent['vout']},]
+        outputs = {self.alice.getnewaddress() : alice_unspent['amount'] - Decimal(0.00030120)} # two outputs
+        alice_txid = self.send_tx(self.alice, inputs, outputs, 3)
+        assert_equal(self.alice.gettransaction(alice_txid)['mempoolconflicts'], [])
+
+    @cleanup
+    def v3_conflict_removed_from_mempool(self):
+        self.log.info("Test a v3 conflict being removed")
+        self.generate(self.nodes[2], 1)
+        # send a v2 output to alice and confirm it
+        txid = self.charlie.sendall([self.alice.getnewaddress()])["txid"]
+        assert_equal(self.charlie.gettransaction(txid, verbose=True)["decoded"]["version"], 2)
+        self.generate(self.nodes[2], 1)
+        # create a v3 tx to alice and bob
+        inputs=[]
+        outputs = {self.alice.getnewaddress() : 2.0, self.bob.getnewaddress() : 2.0}
+        self.send_tx(self.charlie, inputs, outputs, 3)
+
+        alice_v2_unspent = self.alice.listunspent(minconf=1)[0]
+        alice_unspent = [unspent for unspent in self.alice.listunspent(minconf=0) if unspent["confirmations"] == 0][0]
```

Comment:
In 09c9f828250e3840d7557374ff568204305e2679 "test: add truc wallet tests"

`listunspent` also has a `maxconf`, so this could be `self.alice.listunspent(minconf=0, maxconf=0)[0]`